### ISSUE TITLE
25 not fixable vulnerabilities found

### ISSUE THREAD

**Issue by jensgro** [liveblog/liveblog#1281]
I tried multiple times installing and updating the liveblog default-theme. I only get the possibility to install version 3.5.12-beta. And this version has 25 vulnerabilities after installation. I only get rid of two of them by executing `npm audit fix`.

**Comment by jensgro** [comment]
In another attempt npm found 26 not fixable vulnerabilities:

```

                       === npm audit security report ===                        


                                 Manual Review                                  
             Some vulnerabilities require your attention to resolve             
                                                                                
          Visit https://go.npm.me/audit-guide for additional guidance           


  High            Regular Expression Denial of Service                          

  Package         minimatch                                                     

  Patched in      >=3.0.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp > vinyl-fs > glob-stream >      
                  glob > minimatch                                              

  More info       https://npmjs.com/advisories/118                              


  High            Regular Expression Denial of Service                          

  Package         minimatch                                                     

  Patched in      >=3.0.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-hub > gulp > vinyl-fs >         
                  glob-stream > glob > minimatch                                

  More info       https://npmjs.com/advisories/118                              


  High            Regular Expression Denial of Service                          

  Package         minimatch                                                     

  Patched in      >=3.0.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp > vinyl-fs > glob-stream >      
                  minimatch                                                     

  More info       https://npmjs.com/advisories/118                              


  High            Regular Expression Denial of Service                          

  Package         minimatch                                                     

  Patched in      >=3.0.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-hub > gulp > vinyl-fs >         
                  glob-stream > minimatch                                       

  More info       https://npmjs.com/advisories/118                              


  High            Regular Expression Denial of Service                          

  Package         minimatch                                                     

  Patched in      >=3.0.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp > vinyl-fs > glob-watcher >     
                  gaze > globule > glob > minimatch                             

  More info       https://npmjs.com/advisories/118                              


  High            Regular Expression Denial of Service                          

  Package         minimatch                                                     

  Patched in      >=3.0.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-hub > gulp > vinyl-fs >         
                  glob-watcher > gaze > globule > glob > minimatch              

  More info       https://npmjs.com/advisories/118                              


  High            Regular Expression Denial of Service                          

  Package         minimatch                                                     

  Patched in      >=3.0.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp > vinyl-fs > glob-watcher >     
                  gaze > globule > minimatch                                    

  More info       https://npmjs.com/advisories/118                              


  High            Regular Expression Denial of Service                          

  Package         minimatch                                                     

  Patched in      >=3.0.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-hub > gulp > vinyl-fs >         
                  glob-watcher > gaze > globule > minimatch                     

  More info       https://npmjs.com/advisories/118                              


  Moderate        Prototype Pollution                                           

  Package         hoek                                                          

  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-less > less > request > hawk >  
                  boom > hoek                                                   

  More info       https://npmjs.com/advisories/566                              


  Moderate        Prototype Pollution                                           

  Package         hoek                                                          

  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-less > less > request > hawk >  
                  cryptiles > boom > hoek                                       

  More info       https://npmjs.com/advisories/566                              


  Moderate        Prototype Pollution                                           

  Package         hoek                                                          

  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-less > less > request > hawk >  
                  hoek                                                          

  More info       https://npmjs.com/advisories/566                              


  Moderate        Prototype Pollution                                           

  Package         hoek                                                          

  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-less > less > request > hawk >  
                  sntp > hoek                                                   

  More info       https://npmjs.com/advisories/566                              


  Low             Prototype Pollution                                           

  Package         lodash                                                        

  Patched in      >=4.17.5                                                      

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp > vinyl-fs > glob-watcher >     
                  gaze > globule > lodash                                       

  More info       https://npmjs.com/advisories/577                              


  Low             Prototype Pollution                                           

  Package         lodash                                                        

  Patched in      >=4.17.5                                                      

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-hub > gulp > vinyl-fs >         
                  glob-watcher > gaze > globule > lodash                        

  More info       https://npmjs.com/advisories/577                              


  High            Prototype Pollution                                           

  Package         lodash                                                        

  Patched in      >=4.17.11                                                     

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp > vinyl-fs > glob-watcher >     
                  gaze > globule > lodash                                       

  More info       https://npmjs.com/advisories/782                              


  High            Prototype Pollution                                           

  Package         lodash                                                        

  Patched in      >=4.17.11                                                     

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-hub > gulp > vinyl-fs >         
                  glob-watcher > gaze > globule > lodash                        

  More info       https://npmjs.com/advisories/782                              


  High            Prototype Pollution                                           

  Package         lodash                                                        

  Patched in      >=4.17.12                                                     

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp > vinyl-fs > glob-watcher >     
                  gaze > globule > lodash                                       

  More info       https://npmjs.com/advisories/1065                             


  High            Prototype Pollution                                           

  Package         lodash                                                        

  Patched in      >=4.17.12                                                     

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-hub > gulp > vinyl-fs >         
                  glob-watcher > gaze > globule > lodash                        

  More info       https://npmjs.com/advisories/1065                             


  Low             Regular Expression Denial of Service                          

  Package         braces                                                        

  Patched in      >=2.3.1                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-watch > anymatch > micromatch   
                  > braces                                                      

  More info       https://npmjs.com/advisories/786                              


  Low             Regular Expression Denial of Service                          

  Package         braces                                                        

  Patched in      >=2.3.1                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-watch > chokidar > anymatch >   
                  micromatch > braces                                           

  More info       https://npmjs.com/advisories/786                              


  Low             Regular Expression Denial of Service                          

  Package         braces                                                        

  Patched in      >=2.3.1                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > nunjucks > chokidar > anymatch >     
                  micromatch > braces                                           

  More info       https://npmjs.com/advisories/786                              


  Low             Denial of Service                                             

  Package         mem                                                           

  Patched in      >=4.0.0                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-purifycss > purify-css > yargs  
                  > os-locale > mem                                             

  More info       https://npmjs.com/advisories/1084                             


  High            Insufficient Entropy                                          

  Package         cryptiles                                                     

  Patched in      >=4.1.2                                                       

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-less > less > request > hawk >  
                  cryptiles                                                     

  More info       https://npmjs.com/advisories/1464                             


  Moderate        Regular Expression Denial of Service                          

  Package         acorn                                                         

  Patched in      >=6.4.1 <7.0.0 || >=7.1.1                                     

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > browserify > module-deps >           
                  detective > acorn                                             

  More info       https://npmjs.com/advisories/1488                             


  Moderate        Regular Expression Denial of Service                          

  Package         acorn                                                         

  Patched in      >=6.4.1 <7.0.0 || >=7.1.1                                     

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > eslint > espree > acorn              

  More info       https://npmjs.com/advisories/1488                             


  Moderate        Regular Expression Denial of Service                          

  Package         acorn                                                         

  Patched in      >=6.4.1 <7.0.0 || >=7.1.1                                     

  Dependency of   liveblog-default-theme                                        

  Path            liveblog-default-theme > gulp-eslint > eslint > espree >      
                  acorn                                                         

  More info       https://npmjs.com/advisories/1488                             

found 26 vulnerabilities (6 low, 7 moderate, 13 high) in 16934 scanned packages
  26 vulnerabilities require manual review. See the full report for details.

```

**Comment by jensgro** [comment]
I started reviewing the noted vulnerabilities. The tool minimatch exists in version 3.0.4 in my node_modules and therefor should be patched, according to the error message. I don't get it.

**Comment by eos87** [comment]
Hi @jensgro, thanks for reporting this. We plan to fix this in near future but in the meantime if you feel like providing a PR for it, would be more that welcome. We'll be glad to review the PR and merge it if all goes fine :)

**Comment by jensgro** [comment]
@eos87   I tried to fix the problems by installing the mentioned packages manually in a newer version. Unfortunately this doesn't work. And by reviewing my efforts I recognized that some reported problems shouldn't be problems at all. 
As an example: there is a problem mentioned in lodash which should be fixed in versions >= 4.17.12. The lodash version I found in my node_modules is v4.17.15, so there shouldn't be a problem after all.

Nevertheless, this project has the problem that many vulnerabilities are reported that obviously cannot be fixed. Maybe this is a problem new to npm or it is a deeper hidden problem that triggers false flags. 

I try to fix the problem but it is impossible to me. And maybe those vulnerarbilities are the reason why my own modified theme doesn't work at all.
